libraries/tools/kotlin-gradle-plugin-integration-tests/src/test/kotlin/org/jetbrains/kotlin/gradle/ConfigurationAvoidanceIT.kt

```diff
@@ -28,4 +29,46 @@ class ConfigurationAvoidanceIT : BaseGradleIT() {
         }
     }
 
+    @Test
+    fun testAndroidUnrelatedTaskNotConfigured() = with(
+        Project(
+            "AndroidProject",
+            gradleVersionRequirement = GradleVersionRequired.AtLeast("6.6.1")
+        )
+    ) {
+        setupWorkingDir()
+
+        listOf("Android", "Test").forEach { subproject ->
+            gradleBuildScript(subproject).appendText("\n" + """
+                android {
+                    applicationVariants.all {
+                        it.getAidlCompileProvider().configure {
```

Comment:
Based on the bug reported, generateDebugBuildConfig and compileDebugRenderscript are the problems. The creation of compileDebugAidl is caused by square's task, right?

Reply:
Any source-generating task used to be configured, including AIDL. This test (which does not use any third-party plugin) will catch regression if we start configuring them again.

Reply:
Cool, I just want to make sure this test fail without your change which I can confirm now. So this change looks good to me!